core/src/main/java/io/grpc/internal/ServerCallImpl.java

```diff
@@ -121,33 +123,45 @@ public boolean isCancelled() {
     return cancelled;
   }
 
-  ServerStreamListenerImpl newServerStreamListener(ServerCall.Listener<ReqT> listener,
+  ServerStreamListenerImpl<ReqT> newServerStreamListener(ServerCall.Listener<ReqT> listener,
```

Comment:
The generic will need to be propagated to `ServerImpl.startCall()`, `ServerImpl.ServerTransportListenerImpl.streamCreated()`, `ServerImpl.JumpToApplicationThreadServerStreamListener`, and maybe elsewhere.